core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +185,140 @@ export class ContributionAndReviewService {
+
+          const exploration: Exploration = this.explorationObjectFactory.
+            createFromBackendDict(explorationBackendDict);
+          const sortedTranslationSuggestions = (
+            this.sortTranslationSuggestionsByState(
+              fetchSuggestionsResponse.suggestions,
+              exploration.getStates(),
+              explorationBackendDict.init_state_name,
+              this.compareTranslationSuggestions.bind(this)));
+          // eslint-disable-next-line
+          const responseSuggestionIdToDetails: {[key: string]: any} = {};
+          sortedTranslationSuggestions.forEach((suggestion) => {
+            const suggestionDetails = {
+              suggestion: suggestion,
+              details: (
+                fetchSuggestionsResponse.
+                target_id_to_opportunity_dict
+                [suggestion.target_id])
+            };
+            responseSuggestionIdToDetails[
+              suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      // eslint-disable-next-line
+      translationSuggestions: any[],
+      states: States,
+      initStateName: string | null,
+      compareFn: CompareFunction
+      // eslint-disable-next-line
+  ): any[] {
+    // Obtain the state names in the order of content flow in the lesson.
+    if (!initStateName) {
+      return translationSuggestions;
+    }
+
+    const stateNamesInOrder = this.computeGraphService.
+      computeBfsTraversalOfStates(
+        initStateName,
+        states,
+        initStateName
+      );
+    const translationSuggestionsByState = (
+      this.groupTranslationSuggestionsByState(translationSuggestions));
+    const sortedTranslationCards: SuggestionBackendDict[] = [];
+
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = (
+        translationSuggestionsByState.get(stateName) || []);
+      cardsForState.sort(compareFn);
+      translationSuggestionsByState.set(stateName, cardsForState);
+      sortedTranslationCards.push(...cardsForState);
+    }
+    return sortedTranslationCards;
+  }
+
+  // eslint-disable-next-line
+  private groupTranslationSuggestionsByState(
+    translationSuggestions: any[]): Map<string, any[]> {
+    const translationSuggestionsByState = new Map<string, any[]>();
+
+    for (const translationSuggestion of translationSuggestions) {
+      const stateName = translationSuggestion.change.state_name;
+      const suggestionsForState = translationSuggestionsByState.get(stateName) || [];
+      suggestionsForState.push(translationSuggestion);
+      translationSuggestionsByState.set(stateName, suggestionsForState);
+    }
+    return translationSuggestionsByState;
+  }
+
+  // Helper function to get the type order for a given content ID.
```

Comment:
```suggestion
  // Returns the content type order for a given content ID.
```

Nit: You don't need to say "helper function" as it's redundant. Ditto below.